aten/src/ATen/native/cuda/PowKernel.cu

```diff
@@ -22,16 +22,88 @@ static inline __host__ __device__ T powi(T a, T b) {
   return result;
 }
 
+// SFINAE doesn't work well with NVCC under Windows for math functions like pow and sqrt.
+// So we need to define the functions with the explicit function signatures.
+// As for pow, the following signatures are defined as the device function:
+//   pow(float, int)
+//   pow(double, int)
+//   pow(float, float)
+//   pow(double, double)
+// As for sqrt, the following signatures are defined as the device function:
+//   sqrt(float)
+//   sqrt(double)
+// As for inverse sqrt, we must define it explicitly in MSVC, otherwise the static cast will be
+// applied to the result of the inline function, and thus the result is incorrect.
+//   e.g. if we use 1.0 / sqrt(2) for 2 ^ (-0.5) in MSVC, we get
+//          int(2 ^ (-0.5)) = int(1.0 / sqrt(2)) = int(1.0 / int(1.414)) = int(1.0 / 1) = 1
+//        However, the correct result is 
+//          int(2 ^ (-0.5)) = int(1.0 / 1.414) = 0
+#ifdef _MSC_VER
+// Functions for pow
+// pow for at::Half
+static inline __host__ __device__ at::Half pow_(at::Half base, at::Half exp) {
+  return static_cast<at::Half>(std::pow(static_cast<double>(base), static_cast<double>(exp)));
```

Comment:
why do we go to double for float16 precision instead of float32?

Reply:
Let me fix that.